src/test/scala/org/kframework/minikore/NodeTests.scala

```diff
@@ -0,0 +1,205 @@
+package org.kframework.minikore
+
+import org.junit.Test
+import org.kframework.minikore.interfaces.pattern._
+import org.kframework.minikore.interfaces.tree._
+import org.kframework.minikore.implementation.DefaultBuilders
+import org.kframework.minikore.interfaces.build
+
+
+class NodeTest {
+
+  object TestFunctions {
+
+    def size(p: Pattern): Int = p match {
+      case Node(c: Seq[Pattern]) => c.map(size).sum
+      case Leaf(_) => 1
+    }
+
+    def getNode0Count(p: Pattern): Int = p match {
+      case Node0() => 1
+      case Node(c: Seq[Pattern]) => c.map(getNode0Count) sum
```

Comment:
@cos, I do not see `(c map getNode0Count).sum` as a big improvement.  So `c` is a sequence object and map is a member of sequence objects, so on Earth to say `c map f`?  What is wrong with the good old `c.map(f)`?  Also, in all the functional world, `a b c` means `(a b) c`, so why on Earth does Scala need to have `a b c` mean `a.b(c)`?